Sources/OfflineEntitlements/PurchasedProductsFetcher.swift

```diff
@@ -0,0 +1,75 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  PurchasedProductsFetcher.swift
+//
+//  Created by Andrés Boedo on 3/17/23.
+
+import Foundation
+import StoreKit
+
+
+/// This struct should have all the information we need from StoreKit  to create EntitlementInfo from a StoreKit 2 transaction.
+/// Other fields from other places might be needed.
+/// If not, then consider just having a constructor that creates EntitlementInfo from a Storekit.Transaction instead and using
+/// a Factory pattern to store the logic in one place
+@available(iOS 15.0, tvOS 15.0, watchOS 8.0, macOS 12.0, *)
+struct PurchasedSK2Product {
+    let productIdentifier: String
```

Comment:
Nit: line break inside types (it took me a while to get used to this but now it's hard not to hehe)

Reply:
I like it 💪🏻 thanks! FWIW, expect to see some of the ugliest code ever while this is in draft